_posts/2021-04-14-accessing-the-relay-store.markdown

```diff
@@ -0,0 +1,260 @@
+in my eyes is how it tightly couples the client view and API call (in our case, to the GraphQL layer of our stack,
+which we call Metaphysics.) In addition to performance and other benefits, having the data fetch right on top of
+the React component creates a pretty seamless developer experience.
+
+### Building an Artwork Checklist
+
+On PX, we recently launched a checklist feature aimed at empowering our gallery partners to be more self-sufficient
+and find the greatest success possible on Artsy. The checklist prompts galleries to add specific metadata to
+artworks that we know (because of our awesome data team) will make the work more likely to sell. The new feature
+gathers a list of five high-priority artworks (meaning they are published, for-sale, and by a top-selling artist)
+that are missing key pieces of metadata. Users are prompted to add the pieces of meatdata and also have the ability
+to click a button to “snooze” works, which removes them from the list for the day.
+
+<figure class="illustration">
+    <img src="https://user-images.githubusercontent.com/9466631/114630150-28886200-9c77-11eb-9f04-461101496ee0.png">
+</figure>
+
+The feature makes use of [Redis][about-redis], a key-value store used for in-memory cache, to store two lists:
+
+1. `includeIDs` to store the five artworks in the list, so users see a consistent list of artworks whenever they
+   log in and load the page
+2. `excludeIDs` or “snoozed” IDs which Redis will store for 24 hours and ensure the user does not see
+
+When a user presses the “snooze” button, the ID for the artwork is added to the snoozed list in Redis. The list
+`includeIDs` and the list of `excludeIDs` are passed down from Rails controllers to our HAML views and then passed
+as props into our React `HomePageChecklist` app. In our Checklist component, we use both the `includeIDs` and the
+`excludeIDs` as arguments passed to our Relay query to determine what is returned from Metaphysics (Artsy's GraphQL
+layer).
+
+```js
+fragment ArtworksMissingMetadata_partner on Partner
+@argumentDefinitions(
+  first: { type: "Int", defaultValue: 5 }
+  after: { type: "String" }
+  checklistArtworkIDs: { type: "[String!]" }
+  checklistExcludedArtworkIDs: { type: "[String!]" }
+  missingPriorityMetadata: { type: "Boolean", defaultValue: true }
+  publishedWithin: { type: "Int", defaultValue: 7776000 }
+  shallow: { type: "Boolean", defaultValue: false }
+) {
+  id
+  artworksConnection(
+    first: $first
+    after: $after
+    artworkIDs: $checklistArtworkIDs
+    exclude: $checklistExcludedArtworkIDs
+    missingPriorityMetadata: $missingPriorityMetadata
+    publishedWithin: $publishedWithin
+    shallow: $shallow
+  ) @connection(key: "ArtworksMissingMetadata_partner_artworksConnection", filters: []) {
+    totalCount
+    edges {
+      node {
+        internalID
+        ...ArtworksMissingMetadataItem_artwork
+      }
+    }
+  }
+}
+```
+
+## The Damn User
```

Comment:
Since the order of the post is 'problem, first try, second try' this subheading could probably be something like "First try at a solution" (or something). Just some kind of clear division between this section and the next calling out the evolution in thinking/implementation.

Reply:
Also, _I_ know you don't actually hate the user, but I have an aversion to using language that when the humor is removed comes off as hostile, and I'd remove any opportunities for people to point at artsy.github.io and say "see, they hate their users!"

Reply:
Hahaha wow great point @pepopowitz and yes, agree the headings should set a more clear structure